chef/cookbooks/cinder/recipes/volume.rb

```diff
@@ -43,6 +43,8 @@ def make_loopback_file(node, volume)
   max_fsize = ((`df -Pk #{encl_dir}`.split("\n")[1].split(" ")[3].to_i * 1024) * 0.90).to_i rescue 0
   fsize = max_fsize if fsize > max_fsize
 
+  FileUtils.mkdir_p(fdir) unless File.exists?(fdir)
```

Comment:
File.exists? is not really correct. You should check it with Dir, and if i remember correctly exists? have to be replaced with exist? in both cases.